open-metadata-resources/open-metadata-samples/open-metadata-security-samples/src/main/java/org/odpi/openmetadata/metadatasecurity/samples/AbstractCocoPharmaServerSecurityConnector.java

```diff
@@ -0,0 +1,1848 @@
+        zoneSetUp.add(groupMaker.tanyaTidieUserId);
+        zoneAccess.put(clinicalTrialsZoneName, zoneSetUp);
+        zoneSetUp = new ArrayList<>();
+        zoneSetUp.add(groupMaker.faithBrokerUserId);
+        zoneAccess.put(hrZoneName, zoneSetUp);
+        zoneSetUp = new ArrayList<>();
+        zoneSetUp.add(groupMaker.reggieMintUserId);
+        zoneSetUp.add(groupMaker.tomTallyUserId);
+        zoneSetUp.add(groupMaker.sallyCounterUserId);
+        zoneAccess.put(financeZoneName, zoneSetUp);
+        zoneSetUp = new ArrayList<>();
+        zoneSetUp.add(groupMaker.garyGeekeUserId);
+        zoneAccess.put(infrastructureZoneName, zoneSetUp);
+        zoneSetUp = new ArrayList<>();
+        zoneSetUp.add(groupMaker.pollyTaskerUserId);
+        zoneSetUp.add(groupMaker.bobNitterUserId);
+        zoneSetUp.add(groupMaker.lemmieStageUserId);
+        zoneSetUp.add(groupMaker.nancyNoahUserId);
+        zoneSetUp.add(groupMaker.desSignaUserId);
+        zoneAccess.put(developmentZoneName, zoneSetUp);
+        zoneSetUp = new ArrayList<>();
+        zoneSetUp.add(groupMaker.stewFasterUserId);
+        zoneAccess.put(manufacturingZoneName, zoneSetUp);
+        zoneSetUp = new ArrayList<>();
+        zoneSetUp.add(groupMaker.garyGeekeUserId);
+        zoneSetUp.add(groupMaker.erinOverviewUserId);
+        zoneSetUp.add(groupMaker.ivorPadlockUserId);
+        zoneSetUp.add(groupMaker.julesKeeperUserId);
+        zoneSetUp.add(groupMaker.pollyTaskerUserId);
+        zoneSetUp.add(groupMaker.faithBrokerUserId);
+        zoneSetUp.add(groupMaker.reggieMintUserId);
+        zoneAccess.put(governanceZoneName, zoneSetUp);
+    }
+
+    /**
+     * Check that the calling user is authorized to issue a (any) request to the OMAG Server Platform.
+     *
+     * @param userId calling user
+     *
+     * @throws UserNotAuthorizedException the user is not authorized to access this function
+     */
+    @Override
+    public void  validateUserForServer(String   userId) throws UserNotAuthorizedException
+    {
+        if (allUsers.contains(userId))
+        {
+            return;
+        }
+
+        super.validateUserForServer(userId);
+    }
+
+
+    /**
+     * Check that the calling user is authorized to update the configuration for a server.
+     *
+     * @param userId calling user
+     *
+     * @throws UserNotAuthorizedException the user is not authorized to change configuration
+     */
+    @Override
+    public void  validateUserAsServerAdmin(String   userId) throws UserNotAuthorizedException
+    {
+        if (serverAdmins.contains(userId))
+        {
+            return;
+        }
+
+        super.validateUserAsServerAdmin(userId);
+    }
+
+    /**
+     * Abstract declarations to CocoPharmaServerSecurityConnector class
+     * Methods are used to add users to the in-memory groups
+     */
+    public abstract void addUsersToAllUsersGroup() throws NamingException;
+    public abstract void addUsersToAllEmployeesGroup();
```

Comment:
instead of public abstract void addUsersToAllUsersGroup() throws NamingException;
    public abstract void addUsersToAllEmployeesGroup();
    public abstract void addUsersToAssetOnboardingGroup();
    public abstract void addUsersToAssetOnboardingExitGroup();
    public abstract void addUsersToServerAdminsGroup() throws NamingException;
    public abstract void addUsersToServerOperatorsGroup() throws NamingException;
    public abstract void addUsersToServerInvestigatorsGroup() throws NamingException;
    public abstract void addUsersToMetadataArchitectsGroup() throws NamingException;
    public abstract void addUsersToExternalUsersGroup() throws NamingException;
    public abstract void addUsersToNpaAccountsGroup(); (edited) 
14:39
rename from addUsersTo... to retrieveUsersFrom.... and then remove any hard coding of users from the LDAP connector and get those values from LDAP